### PR TITLE
feat(sampling): Onboarding modal step groundwork [TET-164]

### PR DESCRIPTION
**Pull request by matejminar**
https://user-images.githubusercontent.com/9060071/176710013-9511675c-8647-474a-bca4-02f29966f9c2.mp4

### CODE REVIEW

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 1]
```diff
@@ -17,22 +18,26 @@ import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHea
...
 
   const [_rules, setRules] = useState<SamplingRules>([]);
+  const [project, setProject] = useState<Project>();
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState<string | undefined>(undefined);
 
   useEffect(() => {
     async function fetchRules() {
       try {
+        // TODO(sampling): no need to fetch project here, settings pages get it via props for free
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:40]: oh nice! good catch

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 2]
```diff
@@ -72,18 +84,21 @@ export function ServerSideSampling() {
...
-          {error && <LoadingError message={error} />}
-          {!error && loading && <LoadingIndicator />}
-          {!error && !loading && (
-            <RulesPanel
-              headers={['', t('Operator'), t('Condition'), t('Rate'), t('Active'), '']}
-            >
-              <Promo />
-            </RulesPanel>
+        <PermissionAlert
+          access={['project:write']}
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:88]: noticed the same thing https://github.com/getsentry/sentry/blob/b78a5c7789210f157062791859b2df80f97b202f/static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx#L96 :)

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 3]
```diff
@@ -72,18 +84,21 @@ export function ServerSideSampling() {
...
+          message={t(
+            'These settings can only be edited by users with the organization owner, manager, or admin role.'
           )}
-        </Access>
+        />
+        {error && <LoadingError message={error} />}
+        {!error && loading && <LoadingIndicator />}
+        {!error && !loading && (
+          <RulesPanel
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:96]: we will have conflicts here

**Comment by priscilawebdev** [comment]
image

Maybe something for a follow-up, but it seems that the recommended client will have a tooltip

**Review by priscilawebdev — APPROVED** [review]
left some comments, but it looks good to me.

Are we planning to add tests?

File: static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                id="sampling-recommended"
+                checked={selectedStrategy === Strategy.RECOMMENDED}
+                onChange={() => {
+                  setSelectedStrategy(Strategy.RECOMMENDED);
+                }}
+              />
+              {isEdited ? t('New') : t('Recommended')}
+            </Label>
+            <RightAligned>
+              <StyledTextField
```
[the remark this answers is not in the document]
Comment (matejminar) [reply static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx]: Changed to NumberField 👍
Later we might want to do some fancier validation.

**Pull request merged**